crates/astria-conductor/src/block_verifier.rs

```diff
@@ -0,0 +1,583 @@
+        .validators
+        .iter()
+        .try_fold(0u64, |acc, validator| acc.checked_add(validator.voting_power)) else {
+            bail!("total voting power exceeded u64:MAX");
+        };
+
+    let validator_map = validator_set
+        .validators
+        .iter()
+        .map(|v| (&v.address, v)) // address is in bech32
+        .collect::<HashMap<_, _>>();
+
+    let mut commit_voting_power = 0u64;
+    for vote in &commit.signatures {
+        // we only care about votes that are for the Commit.BlockId (ignore absent validators and
+        // votes for nil)
+        if vote.block_id_flag != "BLOCK_ID_FLAG_COMMIT" {
+            continue;
+        }
+
+        // verify validator exists in validator set
+        let validator_address = bech32::encode(
+            "metrovalcons",
+            vote.validator_address.0.to_base32(),
+            Variant::Bech32,
+        )
+        .wrap_err("failed to encode validator address to bech32")?;
+        let Some(validator) = validator_map.get(&validator_address) else {
+            bail!("validator {} not found in validator set", validator_address);
+        };
+
+        // verify address in signature matches validator pubkey
+        let address_from_pubkey = public_key_to_address(&validator.pub_key.key.0)
+            .wrap_err("failed to convert validator public key to address")?;
+        ensure!(
+            address_from_pubkey == validator_address,
+            format!(
+                "validator address mismatch: expected {}, got {}",
+                validator_address, address_from_pubkey
+            )
+        );
+
+        // verify vote signature
+        verify_vote_signature(
+            vote,
+            commit,
+            chain_id,
+            &validator.pub_key.key.0,
+            &vote.signature.0,
+        )
+        .wrap_err("failed to verify vote signature")?;
+
+        commit_voting_power += validator.voting_power;
+    }
+
+    ensure!(
+        commit_voting_power <= total_voting_power,
+        format!(
+            "commit voting power is greater than total voting power: {} > {}",
+            commit_voting_power, total_voting_power
+        )
+    );
+
+    ensure!(
+        does_commit_voting_power_have_quorum(commit_voting_power, total_voting_power),
+        format!(
+            "commit voting power is less than 2/3 of total voting power: {} <= {}",
+            commit_voting_power,
+            total_voting_power * 2 / 3,
+        )
+    );
+
+    Ok(())
+}
+
+fn does_commit_voting_power_have_quorum(commited: u64, total: u64) -> bool {
+    if total < 3 {
+        return commited * 3 > total * 2;
+    }
+
+    commited > total / 3 * 2
+}
+
+// see https://github.com/tendermint/tendermint/blob/35581cf54ec436b8c37fabb43fdaa3f48339a170/types/vote.go#L147
+// TODO: we can change these types (CommitSig and Commit) to be the tendermint types
+// after the other relayer types are updated.
+fn verify_vote_signature(
```

Comment:
Nit: Why does this function have no whitespace between statements? I would prefer if our lints were consistent